framework/db/mssql/QueryBuilder.php

```diff
@@ -122,12 +122,16 @@ public function alterColumn($table, $column, $type)
      * Builds a SQL statement for enabling or disabling integrity check.
      * @param boolean $check whether to turn on or off the integrity check.
      * @param string $schema the schema of the tables. Defaults to empty string, meaning the current or default schema.
-     * @param string $table the table name. Defaults to empty string, meaning that no table will be changed.
+     * @param string $table the table name. Defaults to empty string, meaning that all tables will be changed.
      * @return string the SQL statement for checking integrity
      * @throws InvalidParamException if the table does not exist or there is no sequence associated with the table.
      */
     public function checkIntegrity($check = true, $schema = '', $table = '')
     {
+        $enable = $check ? 'CHECK' : 'NOCHECK';
+        if ($table === '') {
+    		return "EXEC sp_MSforeachtable 'ALTER TABLE ? {$enable} CONSTRAINT ALL'";
```

Comment:
this seems to be an unsupported API, is there a way to do it in an officially supported way?

>  It is our general policy not to provide documentation to internal unsupported API like this one.
> https://connect.microsoft.com/SQLServer/feedback/details/264676/sp-msforeachtable-provide-supported-documented-version